File: Store/src/Sales.Domain/Model/Discounts/ValueBasedDiscount.cs
using System;

namespace Sales.Domain.Model.Discounts
{
    public class ValueBasedDiscount : DiscountCalculation
    {
        private readonly int _discountValue;
        public ValueBasedDiscount(int discountValue)
        {
            _discountValue = discountValue;
        }

        public override long CalculateDiscount(long totalPrice)
        {
            throw new NotImplementedException();
        }
    }
}

File: src/main/java/store/sales/domain/model/discounts/ValueBasedDiscount.java
package store.sales.domain.model.discounts;

public class ValueBasedDiscount extends DiscountStrategy {

    private final int discountValue;

    public ValueBasedDiscount(int discountValue) {
        this.discountValue = discountValue;
    }

    @Override
    public int calculateDiscount(int totalPrice) {
        throw new UnsupportedOperationException();
    }
}
